matrix-room.c

```diff
@@ -1163,6 +1237,30 @@ void matrix_room_send_image(PurpleConversation *conv, int imgstore_id,
 }
 
 /**
+ * Sends a typing notification in a room with a 25s timeout
+ */
+void matrix_room_send_typing(PurpleConversation *conv, gboolean typing)
+{
+    JsonObject *content;
+    MatrixConnectionData *acct;
+    PurpleConnection *pc = conv->account->gc;
+
+    acct = purple_connection_get_protocol_data(pc);
+    
+    content = json_object_new();
```

Comment:
the abstraction used elsewhere is for the `matrix_api_*` functions to take the inputs to build the json - in this case, `typing` and `timeout` - so that we constrain the json wrangling to `matrix-api.c` as far as possible.

just fyi really